### PR TITLE
Activity feed screen

### PR DESCRIPTION
**Pull request by sunnyVolkova**
Activity Feed Screen Created

### CODE REVIEW

File: ObjCTraining/ActivityFeedTableViewCell+ConfigureForFeed.m [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- (void)setCellValuesWithFeed:(LCFeed *)feed {
+    if (feed == nil) {
+        NSLog(@"feed is nil");
+        return;
+    }
+    UIImage *image = [UIImage imageNamed:feed.imageName];
+    self.actionImageView.image = image;
+
+    self.descriptionTextLabel.text = feed.descriptionText;
+    self.timeLabel.text = [self getTimeStringFromDate:feed.createdDate];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedTableViewCell+ConfigureForFeed.m]: self.timeLabel.text = [NSDateFormatter localizedStringFromDate:feed.createdDate dateStyle:NSDateFormatterNoStyle timeStyle:NSDateFormatterMediumStyle];

File: ObjCTraining/ActivityFeedTableViewCell.m [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by RWuser on 04/04/16.
+//  Copyright © 2016 RWuser. All rights reserved.
+//
+
+#import "ActivityFeedTableViewCell.h"
+
+@implementation ActivityFeedTableViewCell
+
+- (void)awakeFromNib {
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedTableViewCell.m]: Need to remove empty methods

File: ObjCTraining/ActivityFeedTableViewCell.m [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    // Initialization code
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+
+    // Configure the view for the selected state
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedTableViewCell.m]: Need remove also

File: ObjCTraining/ActivityFeedViewController.m [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+static int const sectionHeaderHeight = 46;
+
+@interface ActivityFeedViewController ()
+@property NSMutableArray *feeds;
+@end
+
+@implementation ActivityFeedViewController
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.estimatedRowHeight = cellHeight;
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m:26]: why do you need this for constant cells's height?

File: ObjCTraining/ActivityFeedViewController.m [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    [self.tableView registerNib:[UINib nibWithNibName:@"ActivityFeedTableViewCell" bundle:nil] forCellReuseIdentifier:activityFeedCellIdentifier];
+    [self initDataSource];
+    self.tableView.dataSource = self;
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: you can set datasource and delegate with interface builder. It's a good idea if you have one table on vc

File: ObjCTraining/ActivityFeedViewController.m [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return self.feeds.count;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return ((NSMutableArray *) (self.feeds[section])).count;
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: The needless space

File: ObjCTraining/ActivityFeedViewController.m [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    ActivityFeedTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:activityFeedCellIdentifier forIndexPath:indexPath];
+    NSArray *lcFeedSection = [self.feeds objectAtIndex:indexPath.section];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: we don't use prefix for var names

File: ObjCTraining/ActivityFeedViewController.m [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    LCFeed *lcFeed = [lcFeedSection objectAtIndex:indexPath.row];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: Name it just "feed"

File: ObjCTraining/ActivityFeedViewController.m [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+    return cell;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
+    return UITableViewAutomaticDimension;
+}
+
+- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section {
+    UILabel *label = [[UILabel alloc] initWithFrame:CGRectZero];
+    [label setTextAlignment:NSTextAlignmentCenter];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: use property instead of sending messages

File: ObjCTraining/ActivityFeedViewController.m [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    [label setTextColor:[UIColor colorWithRed:50.0f/255.0f green:50.0f/255.0f blue:50.0f/255.0f alpha:1.0f]];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: the same

File: ObjCTraining/ActivityFeedViewController.m [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    [label setBackgroundColor:[UIColor colorWithRed:146.0f/255.0f green:212.0f/255.0f blue:250.0f/255.0f alpha:1.0f]];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: the same

File: ObjCTraining/ActivityFeedViewController.m [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+    [label setFont:[UIFont systemFontOfSize:17]];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: the same

File: ObjCTraining/ActivityFeedViewController.m [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+    NSArray *lcFeedSection = [self.feeds objectAtIndex:section];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: "feedSections" sounds better

File: ObjCTraining/ActivityFeedViewController.m [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+    if (lcFeedSection.count > 0) {
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: It's better to use [feedSections firstObject]

File: ObjCTraining/ActivityFeedViewController.m [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+        LCFeed *lcFeed = [lcFeedSection objectAtIndex:0];
+        NSDate *date = lcFeed.createdDate;
+        label.text = [self getDateStringFromDate:date];
+    }
+    return label;
+}
+
+- (UIView *)tableView:(UITableView *)tableView viewForFooterInSection:(NSInteger)section {
+    UIView *view = [[UIView alloc] initWithFrame:CGRectZero];
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: why do we need footer?

File: ObjCTraining/ActivityFeedViewController.m [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+    view.backgroundColor = [UIColor colorWithWhite:0.0 alpha:0.0];
+    return view;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
+    return sectionHeaderHeight;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section {
+    return 0.1f; //can't set to 0.0
```
Comment (YGeorge) [inline ObjCTraining/ActivityFeedViewController.m]: why do you need that?

File: ObjCTraining/ActivityFeedViewController.m (hunk 4 above)
Comment (sunnyVolkova) [inline ObjCTraining/ActivityFeedViewController.m:26]: need for correct displaying cells with variable height

**Pull request merged**